code/Gamemodes/Gamemode.cs

```diff
@@ -147,4 +150,9 @@ internal virtual void OnClientDisconnect( IClient cl, NetworkDisconnectionReason
 	internal virtual void MoveToSpawnpoint( IClient client ) { }
 
 	internal virtual void UseTurn( bool giveMovementGrace = false ) { }
+
+	internal virtual void PlayerJoinedLate( Player player )

```

Comment:
This could be refactored to call OnClientJoined on the late client. We check if the state is playing, and then queue them for being parachuted in. If you disagree, then at the very least rename this method to OnPlayerJoined. It seems like the organization for client re-joining and new players joining should overlap to me, or at least be called from the same place.

Reply:
I don't necessarily disagree. I just figured that there might be more complex logic for handling latejoiners, say, in the team based modes, and figured it was cleaner to have a function dedicated for it. What do you think?